### ISSUE TITLE
gui-client errors during make 

### ISSUE THREAD

**Issue by kiat** [hobbit-project/platform#13]
Hi, 

when I run make, I get lots of these errors. 

I need to get the platform running on my localhost, do I need to have GUI?

Thanks!

```

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/common/src/pipes/async_pipe.d.ts:44:37 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/aot/compiler.d.ts:32:37 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/aot/compiler_host.d.ts:20:32 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/directive_normalizer.d.ts:38:71 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/directive_normalizer.d.ts:40:73 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/i18n/extractor.d.ts:14:32 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/i18n/extractor.d.ts:22:34 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/jit/compiler.d.ts:44:48 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/jit/compiler.d.ts:46:64 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/metadata_resolver.d.ts:58:103 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/resource_loader.d.ts:13:22 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/util.d.ts:33:17 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/compiler/src/util.d.ts:34:45 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/application_init.d.ts:16:17 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/application_ref.d.ts:116:75 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/application_ref.d.ts:132:109 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/application_ref.d.ts:158:66 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/application_ref.d.ts:160:100 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/facade/lang.d.ts:12:16 
Cannot find name 'Map'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/facade/lang.d.ts:13:16 
Cannot find name 'Set'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/linker/compiler.d.ts:53:48 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/linker/compiler.d.ts:61:64 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/linker/ng_module_factory_loader.d.ts:14:33 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/linker/system_js_ng_module_factory_loader.d.ts:28:24 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/core/src/util/lang.d.ts:12:52 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/platform-browser-dynamic/src/resource_loader/resource_loader_impl.d.ts:10:22 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/config.d.ts:307:84 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/interfaces.d.ts:78:98 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/interfaces.d.ts:157:108 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/interfaces.d.ts:227:114 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/interfaces.d.ts:297:88 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/interfaces.d.ts:367:49 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/router.d.ts:392:69 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/router.d.ts:414:58 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/router_module.d.ts:186:22 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@angular/router/src/utils/collection.d.ts:36:78 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@ng2-dynamic-forms/core/src/model/dynamic-form-control-relation.model.ts:24:21 
Property 'find' does not exist on type 'DynamicFormControlRelationGroup[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@ng2-dynamic-forms/core/src/model/dynamic-form-control-relation.model.ts:28:21 
Property 'find' does not exist on type 'DynamicFormControlRelationGroup[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@ng2-dynamic-forms/core/src/model/dynamic-form-control-relation.model.ts:32:21 
Property 'find' does not exist on type 'DynamicFormControlRelationGroup[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@ng2-dynamic-forms/core/src/service/dynamic-form-autofill.service.ts:130:21 
Property 'startsWith' does not exist on type 'string'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@ng2-dynamic-forms/core/src/service/dynamic-form.service.ts:102:21 
Property 'find' does not exist on type 'DynamicFormControlModel[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@ng2-dynamic-forms/core/src/utils.ts:38:37 
Property 'assign' does not exist on type 'ObjectConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:47:35 
Cannot find name 'Iterable'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:350:47 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:351:51 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:352:33 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:353:33 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:354:33 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:355:60 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:356:59 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:357:64 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:357:96 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:358:59 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:359:57 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:360:58 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:361:57 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:362:60 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:362:116 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:363:41 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:365:80 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:366:77 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:367:75 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:368:72 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:431:5 
Cannot find name 'Symbol'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:456:38 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:464:58 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:465:58 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:467:47 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:468:68 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:472:39 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:474:47 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:494:54 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:495:44 
Cannot find name 'PropertyKey'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:511:45 
Cannot find name 'Iterable'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:512:42 
Cannot find name 'Iterable'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:627:17 
Cannot find name 'PromiseConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:642:31 
Cannot find name 'Iterable'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:646:38 
Cannot find name 'Iterable'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:646:52 
Cannot find name 'Iterator'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:657:34 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1280:33 
Property 'for' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1284:40 
Property 'hasInstance' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1288:47 
Property 'isConcatSpreadable' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1292:37 
Property 'iterator' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1296:35 
Property 'keyFor' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1300:34 
Property 'match' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1304:36 
Property 'replace' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1308:35 
Property 'search' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1312:36 
Property 'species' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1316:34 
Property 'split' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1320:40 
Property 'toPrimitive' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1324:40 
Property 'toStringTag' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:1328:40 
Property 'unscopables' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2059:33 
Property 'for' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2063:40 
Property 'hasInstance' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2067:47 
Property 'isConcatSpreadable' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2071:37 
Property 'iterator' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2075:35 
Property 'keyFor' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2079:34 
Property 'match' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2083:36 
Property 'replace' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2087:35 
Property 'search' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2091:36 
Property 'species' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2095:34 
Property 'split' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2099:40 
Property 'toPrimitive' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2103:40 
Property 'toStringTag' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/@types/core-js/index.d.ts:2107:40 
Property 'unscopables' does not exist on type 'SymbolConstructor'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/rxjs/Observable.d.ts:69:59 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/rxjs/Observable.d.ts:69:69 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/rxjs/observable/PromiseObservable.d.ts:40:30 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/rxjs/observable/PromiseObservable.d.ts:41:25 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/rxjs/operator/toPromise.d.ts:2:59 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/rxjs/operator/toPromise.d.ts:3:78 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/node_modules/rxjs/operator/toPromise.d.ts:3:88 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/benchmark-submit.comp.ts:41:50 
Property 'find' does not exist on type 'BenchmarkLight[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/benchmark-submit.comp.ts:56:59 
Property 'find' does not exist on type 'System[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/benchmark-submit.comp.ts:66:59 
Property 'find' does not exist on type 'System[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/challenge-task-edit.comp.ts:50:44 
Property 'findIndex' does not exist on type 'ChallengeTask[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/challenge-task-edit.comp.ts:85:50 
Property 'find' does not exist on type 'BenchmarkLight[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/challenge-tasks-experiments.comp.ts:31:41 
Property 'find' does not exist on type 'ExperimentCount[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/experiments-details.comp.ts:57:28 
Property 'find' does not exist on type 'ConfigurationParameterValue[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/experiments-details.comp.ts:70:21 
Cannot find name 'Map'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/experiments-details.comp.ts:71:27 
Cannot find name 'Map'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/model.ts:65:22 
Property 'includes' does not exist on type 'string[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/navbar.comp.ts:29:36 
Property 'findIndex' does not exist on type 'string[]'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/services/customHttp.ts:56:12 
Property 'startsWith' does not exist on type 'string'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/services/customHttp.ts:82:12 
Property 'startsWith' does not exist on type 'string'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/services/keycloakService.ts:12:17 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/services/keycloakService.ts:17:22 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/services/keycloakService.ts:70:14 
Cannot find name 'Promise'.

ERROR in [default] /home/kia/Desktop/platform/hobbit-gui/gui-client/src/app/services/keycloakService.ts:72:15 
Cannot find name 'Promise'.

```

**Comment by MichaelRoeder** [comment]
1. Do you have checked the [platform requirements](https://github.com/hobbit-project/platform#requirements)?
2. Which command creates the errors? There are two commands executed for the GUI-client, `npm install` and `npm run build`. Normally, make should print the command that exited with an error code. Without knowing the command that creates these errors, it is hard to help.

**Comment by kiat** [comment]
I need just to run the platform on my local machine. I think that GUI-Client is not important for me, I should be able to run the platform without it. 

When I run 'npm run build' 

I get this 
```
npm run build 
npm ERR! path /home/kia/Desktop/platform/package.json
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall open
npm ERR! enoent ENOENT: no such file or directory, open '/home/kia/Desktop/platform/package.json'
npm ERR! enoent ENOENT: no such file or directory, open '/home/kia/Desktop/platform/package.json'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kia/.npm/_logs/2017-03-20T19_22_09_789Z-debug.log
```

and inside the log file I have 

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/local/bin/npm', 'run', 'build' ]
2 info using npm@4.4.1
3 info using node@v4.2.6
4 verbose stack Error: ENOENT: no such file or directory, open '/home/kia/Desktop/platform/package.json'
4 verbose stack     at Error (native)
5 verbose cwd /home/kia/Desktop/platform
6 verbose Linux 4.4.0-66-generic
7 verbose argv "/usr/bin/nodejs" "/usr/local/bin/npm" "run" "build"
8 verbose node v4.2.6
9 verbose npm  v4.4.1
10 error path /home/kia/Desktop/platform/package.json
11 error code ENOENT
12 error errno -2
13 error syscall open
14 error enoent ENOENT: no such file or directory, open '/home/kia/Desktop/platform/package.json'
15 error enoent ENOENT: no such file or directory, open '/home/kia/Desktop/platform/package.json'
15 error enoent This is most likely not a problem with npm itself
15 error enoent and is related to npm not being able to find a file.
16 verbose exit [ -2, true ]
```

Could you please check your installation/running instructions on fresh installed Ubuntu 16.04 

Also I have some troubles with the storage configurations. 
I installed virtuoso, and set up the users and passwords. 

but when running  **docker-compose up virtuoso**

I get 

```
docker-compose up virtuoso
Starting vos

ERROR: for virtuoso  Cannot start service virtuoso: driver failed programming external connectivity on endpoint vos (d2861f3e75deb4377e5a8b2461505c4b880d50f6dee708f627b2178b214c38ef): Error starting userland proxy: listen tcp 0.0.0.0:8890: bind: address already in use
ERROR: Encountered errors while bringing up the project.
```

**Comment by MichaelRoeder** [comment]
Sorry for making the following step not clear from the beginning on. Please execute the two npm commands again inside the `platform/hobbit-gui/gui-client` directory.

Starting the virtuoso fails because the port `8890` is already in use on your machine. Please open the `docker-compose.yml` file in the `platform` directory and scroll to this line: https://github.com/hobbit-project/platform/blob/master/docker-compose.yml#L87
If you don't want to use the SPARQL endpoint of your local platform, you can comment out the line using `#`. Otherwise, you should change the left `8890` to a free port on your local machine at which the virtuoso will be accessible.

**Comment by MichaelRoeder** [comment]
@kiat does the problem still exist? Did you tried to execute the two `npm` commands in the `platform/hobbit-gui/gui-client` directory?

**Comment by MichaelRoeder** [comment]
@kiat Can you please retry it using the latest version (v1.0.4)? It should be fixed now.

**Comment by MichaelRoeder** [comment]
The problem should be fixed since v1.0.4
